src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -220,6 +220,9 @@ StartMetadataSyncToNode(const char *nodeNameString, int32 nodePort)
 
 	SyncMetadataSnapshotToNode(workerNode, raiseInterrupts);
 	MarkNodeMetadataSynced(workerNode->workerName, workerNode->workerPort, true);
```

Comment:
Few questions:

1) Do we need `MarkNodeHasMetadata` and `MarkNodeMetadataSynced` functions any more at this functio and stop_metadata? `SetWorkerColumn` already does what they do?

2) What about `SyncMetadataToNodes`? When that function syncs the metadata, we do not seem to propagate the changes to the workers. Can we replace `MarkNodeMetadataSynced` to `SetWorkerColumn` there? 

3) Should we do `SetWorkerColumn(.. Anum_pg_dist_node_metadatasynced)` in `SyncMetadataToNodes` as well?

4) If we do all 1,2 and 3 above, can we remove `MarkNodeMetadataSynced` function at all? I think every caller should use `SetWorkerColumn`, because we always want to sync the changes to the nodes

Reply:
1.Makes sense. I've added these changes in my last commits.

2&3.  It seems these functions are not meant to be used together.
```
/*
 * SyncMetadataToNodes tries recreating the metadata snapshot in the
 * metadata workers that are out of sync. Returns the result of
 * synchronization.
 */
```
It seems `SyncMetadataToNodes` targets only the metadata nodes that are out of sync. However, `SetWorkerColumn` brings us to an error if a metadata node is out of sync. So calling it in `SyncMetadataToNodes` gives us nothing but errors. How about changing `SetWorkerColumn` so that it doesn't error out in case of any "not synced metadata nodes" ? But this could affect many different places where we use this function and also `SendCommandToWorkersWithMetadata`.